metrics/sinks/influxdb/influxdb.go

```diff
@@ -257,17 +257,49 @@ func (sink *influxdbSink) createDatabase() error {
 	if sink.dbExists {
 		return nil
 	}
+
+	// check whether the specific database exists or not
 	q := influxdb.Query{
-		Command: fmt.Sprintf(`CREATE DATABASE %s WITH NAME "default"`, sink.c.DbName),
+		Command: "SHOW DATABASES",
 	}
 
-	if resp, err := sink.client.Query(q); err != nil {
-		if !(resp != nil && resp.Err != nil && strings.Contains(resp.Err.Error(), "already exists")) {
-			err := sink.createRetentionPolicy()
-			if err != nil {
-				return err
+	resp, err := sink.client.Query(q)
+
+	if err != nil {
+		return err
+	}
+
+	if resp != nil && len(resp.Results) == 1 && len(resp.Results[0].Series) == 1 {
+		values := resp.Results[0].Series[0].Values
+		for _, name := range values {
+			if sink.c.DbName == name[0].(string) {
+				sink.dbExists = true
+				//If the database exists,we should exit right now.
```

Comment:
`//If the database exists,we should exit right now.` -> `// If the database exists, we should exit right now.`